product/validators/purchase_request_validator.py

```
@@ -0,0 +1,15 @@
+from product.exceptions import ItemNotFoundException, InactiveItemException
+from product.models.item import Item
+from product.services.banned_item_checker import check_item_banned
+
+
+def validate_purchase_request(item_id):
+    try:
+        item = Item.objects.get(id=item_id)
+    except Item.DoesNotExist:
+        raise ItemNotFoundException()
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)